compiler/filedescriptor.go

```diff
@@ -180,7 +195,49 @@ func jsonStr(s string) *string {
 }
 
 func newService(s *parser.Service, scope []string, types types) *pb.ServiceDescriptorProto {
-	panic(fmt.Sprintf("not implemented: newSeervice %v %v %v", s, scope, types))
+	methods := []*pb.MethodDescriptorProto{}
+	for _, e := range s.Entry {
+		if e.Method != nil {
+			method := newMethod(e.Method, scope, types)
+			methods = append(methods, method)
+		}
+	}
+
+	sd := &pb.ServiceDescriptorProto{
+		Name:    &s.Name,
+		Method:  methods,
+		Options: nil,
+	}
+
+	return sd
+}
+
+func newMethod(m *parser.Method, scope []string, types types) *pb.MethodDescriptorProto {
+	var clientStreaming, serverStreaming *bool
+	if m.StreamingRequest {
+		clientStreaming = &m.StreamingRequest
+	}
+	if m.StreamingResponse {
+		serverStreaming = &m.StreamingResponse
+	}
+	inputEnum, inputType := newFieldDescriptorProtoType(m.Request, scope, types)
+	if inputEnum != pb.FieldDescriptorProto_TYPE_MESSAGE {
+		panic(fmt.Sprintf(" e.Pos: method %s should have Message as request type", m.Name))
```

Comment:
Is `e.Pos` meant to be displayed here and below instead?

Reply:
fixed.